code-challenges.js

```diff
@@ -52,9 +112,25 @@ var notAString2 = 42
 // Expected output: [ { name: "Stimpy", animal: "cat" }, { name: "Scratchy", animal: "cat" }, { name: "Felix", animal: "cat" } ]
 
 
-var toonimals = [ {name: "Itchy", animal: "mouse"}, {name: "Stimpy", animal: "cat"}, {name: "Daffy", animal: "duck"}, {name: "Scratchy", animal: "cat"}, {name: "Ren", animal: "dog"}, {name: "Felix", animal: "cat"}]
+ var toonimals = [ {name: "Itchy", animal: "mouse"}, {name: "Stimpy", animal: "cat"}, {name: "Daffy", animal: "duck"}, {name: "Scratchy", animal: "cat"}, {name: "Ren", animal: "dog"}, {name: "Felix", animal: "cat"}]
 
+ //declare a function that takes an argument
+  const cats = (array) => {
+  // create a higher order function that only returns only the toon object that are cats
+   return array.filter(value => value.animal === "cat")
+  }
+  //call the function and run it in the terminal
+  console.log(cats(toonimals))
 
 
 // --------------------7) Using the toonimals variable from #6, create a function that returns only the names of the non-cats.
 // Expected output: "Itchy" "Daffy" "Ren"
+
+var toonimals = [ {name: "Itchy", animal: "mouse"}, {name: "Stimpy", animal: "cat"}, {name: "Daffy", animal: "duck"}, {name: "Scratchy", animal: "cat"}, {name: "Ren", animal: "dog"}, {name: "Felix", animal: "cat"}]
+//declare a function that takes an argument
+ const nonCats = (array) => {
+   // create a higher order function that only returns only the toon objects that are not cats
+   return array.filter(value => value.animal !== "cat")
+}
```

Comment:
this solution is _suuuper_ close! we just need to add a little more logic so that we only get the **names** of the cat in the return. hint: you'll want to add a map as well as the filter